src/include/sof/audio/module_adapter/module/module_interface.h

```diff
@@ -143,6 +143,14 @@ struct module_interface {
 	 * free in .free(). This should free all memory allocated during module initialization.
 	 */
 	int (*free)(struct processing_module *mod);
+	/**
+	 * Module specific bind procedure, called when modules are bound with each other
```

Comment:
Should have some documentation on what "data" is and what the conditions are (is the new state reflected in module internals or not).